AX: Add MessageSource::Accessibility console messages for Web Inspector

rahimabdi opened the pull request:
#### b9845849e6d9330cf56ca8d9df2665ec6dad16d0
<pre>
AX: Add MessageSource::Accessibility console messages for Web Inspector
<a href="https://bugs.webkit.org/show_bug.cgi?id=290157">https://bugs.webkit.org/show_bug.cgi?id=290157</a>

Reviewed by Tyler Wilcock.

To facilitate logging of accessibility-specific author errors (such as aria-hidden set
on a focusable element), this PR adds an &quot;Accessibility&quot; token to MessageSource for Web
Inspector console logging.

Source/JavaScriptCore:

* runtime/ConsoleTypes.h:
* runtime/ConsoleClient.cpp:
(JSC::appendMessagePrefix):
Add `Accessibility` to MessageSource.
* inspector/ConsoleMessage.cpp:
(Inspector::messageSourceValue):
Add `Accessibility` message source.
* inspector/protocol/Console.json:
(&quot;enum&quot; key)
Add &quot;accessibility&quot;.

Source/WebInspectorUI:

* UserInterface/Models/ConsoleMessage.js:
* UserInterface/Models/IssueMessage.js:
(WI.IssueMessage):
Add `Accessibility` message source.

Source/WebKitLegacy/mac:

* WebCoreSupport/WebChromeClient.mm:
(stringForMessageSource):
Add `MessageSource::Accessibility` message source.

Canonical link: <a href="https://commits.webkit.org/292838@main">https://commits.webkit.org/292838@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3739ddf2b665b836f0721d7056aa2a77bae41845

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97239 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16865 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7079 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102324 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47768 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/99284 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17158 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25312 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74093 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31287 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100242 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12979 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87943 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54430 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12733 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5818 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47209 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/89916 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82780 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5903 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104346 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/95862 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24315 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/17736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83139 "Passed tests") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/24690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/84068 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82547 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20770 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4774 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/17830 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24281 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29434 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/119487 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24103 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33546 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27417 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25677 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

twilco reviewed (approved):
Looks good to me, but would be nice to have a Web Inspector expert take a look too.

rcaliman-apple commented on Source/JavaScriptCore/inspector/ConsoleMessage.cpp:
```diff
@@ -192,6 +192,7 @@ static Protocol::Console::ChannelSource messageSourceValue(MessageSource source)
     case MessageSource::AppCache: return Protocol::Console::ChannelSource::Appcache;
     case MessageSource::Rendering: return Protocol::Console::ChannelSource::Rendering;
     case MessageSource::CSS: return Protocol::Console::ChannelSource::CSS;
+    case MessageSource::Accessibility: return Protocol::Console::ChannelSource::Accessibility;
```
We should also add the Accessibility channel source in `PageConsoleAgent::getLoggingChannels()` https://github.com/WebKit/WebKit/blob/49ef75665309c974a2c50b42923c58f898652d7f/Source/WebCore/inspector/agents/page/PageConsoleAgent.cpp#L91-L107

It won't be used in the Web Inspector UI now, but it should be available if we want to use it in the future.

For reference, only `Media`, `WebRTC` and `MediaSource` are currently used to filter the Console:

https://github.com/WebKit/WebKit/blob/325521c03de8835c85c51dca6ddffa62c254beff/Source/WebInspectorUI/UserInterface/Views/LogContentView.js#L112-L114

And only for these channel sources do we support configuring the log level:

https://github.com/WebKit/WebKit/blob/325521c03de8835c85c51dca6ddffa62c254beff/Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js#L373-L382

rahimabdi replied:
@rcaliman-apple Added, thanks.

rcaliman-apple commented on Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
```diff
@@ -404,6 +405,8 @@ - (void)setIsSelected:(BOOL)isSelected;
         return WebConsoleMessageRenderingMessageSource;
     case MessageSource::CSS:
         return WebConsoleMessageCSSMessageSource;
+    case MessageSource::Accessibility:
```
@dcrousso Do we have to support this in WebKitLegacy?

rahimabdi replied:
@rcaliman-apple Should I remove this?

rcaliman-apple replied:
We don't have to add new support to `WebKitLegacy`, but should not cause it to crash / fail existing tests. 

You can try to remove this change, push a patch, then monitor the EWS table in the PR description for any `mac-wk1` failures. If it's green, I think it's ok to remove.

rahimabdi replied:
@rcaliman-apple I'll leave as is, thanks.

rcaliman-apple reviewed (commented):
LGTM 👍 

Left a comment about adding the new channel source in `PageConsoleAgent::getLoggingChannels()`

rcaliman-apple commented on Source/WebCore/inspector/agents/page/PageConsoleAgent.cpp:
```diff
@@ -96,6 +96,7 @@ Inspector::Protocol::ErrorStringOr<Ref<JSON::ArrayOf<Inspector::Protocol::Consol
     addLogChannel(Inspector::Protocol::Console::ChannelSource::Appcache);
     addLogChannel(Inspector::Protocol::Console::ChannelSource::Rendering);
     addLogChannel(Inspector::Protocol::Console::ChannelSource::CSS);
+    addLogChannel(Inspector::Protocol::Console::ChannelSource::Accessibility);
```
The test failure of [inspector/console/webcore-logging.html](https://ews-build.s3-us-west-2.amazonaws.com/macOS-Sonoma-Release-WK2-Intel-Tests-EWS/1c9cb2b8-4248/results.html) with the latest patch is related to adding this new channel. 

Seems like you just have to update the test expectation file:

```
run-webkit-tests --no-build --reset-results inspector/console/webcore-logging.html
```

rahimabdi replied:
Done, thanks @rcaliman-apple!

burg reviewed (changes requested):
Code change looks OK.

Please rebaseline the test results as Razvan says. Then we can get this merged. 👍🏼

rahimabdi committed:
AX: Add MessageSource::Accessibility console messages for Web Inspector
https://bugs.webkit.org/show_bug.cgi?id=290157

Reviewed by Tyler Wilcock.

To facilitate logging of accessibility-specific author errors (such as aria-hidden set
on a focusable element), this PR adds an "Accessibility" token to MessageSource for Web
Inspector console logging.

Source/JavaScriptCore:

* runtime/ConsoleTypes.h:
* runtime/ConsoleClient.cpp:
(JSC::appendMessagePrefix):
Add `Accessibility` to MessageSource.
* inspector/ConsoleMessage.cpp:
(Inspector::messageSourceValue):
Add `Accessibility` message source.
* inspector/protocol/Console.json:
("enum" key)
Add "accessibility".

Source/WebInspectorUI:

* UserInterface/Models/ConsoleMessage.js:
* UserInterface/Models/IssueMessage.js:
(WI.IssueMessage):
Add `Accessibility` message source.

Source/WebKitLegacy/mac:

* WebCoreSupport/WebChromeClient.mm:
(stringForMessageSource):
Add `MessageSource::Accessibility` message source.

Canonical link: https://commits.webkit.org/292838@main
